### PR TITLE
fix(l10n): Localize ModalVerifySession component

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- The ModalVerifySession component was not localized

## This pull request

- Localizes that component
- I also had to add some localization to the base `Modal` component to handle the tooltip

## Issue that this pull request solves

Closes: #7940 

## Checklist

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-settings/src/components/Modal/index.tsx [hunk 1]
```diff
@@ -68,7 +70,7 @@ export const Modal = ({
             <button
               data-testid="modal-dismiss"
               onClick={(event) => onDismiss()}
-              title="Close modal"
+              title={l10n.getString('modal-close-tooltip', null, 'Close modal')}
```
Comment (dannycoates) [inline packages/fxa-settings/src/components/Modal/index.tsx:73]: `modal-close-title`?

File: packages/fxa-settings/src/components/Modal/en-US.ftl [hunk 2]
```diff
@@ -0,0 +1,4 @@
+## Modal
+
+modal-close-title = Close modal
```
Comment (dannycoates) [inline packages/fxa-settings/src/components/Modal/en-US.ftl:3]: I wonder if we can do better than "Close modal"? Do most folks know what "modal" means?

File: packages/fxa-settings/src/components/Modal/index.tsx (hunk 1 above)
Comment (vbudhram) [reply packages/fxa-settings/src/components/Modal/index.tsx:73]: >.< good catch

File: packages/fxa-settings/src/components/ModalVerifySession/index.tsx [hunk 3]
```diff
@@ -109,26 +112,41 @@ export const ModalVerifySession = ({
...
-          Verify your email
-        </h2>
-        <p
-          id="modal-verify-session-desc"
-          data-testid="modal-desc"
-          className="my-6 text-center"
+        <Localized id="mvs-verify-your-email">
+          <h2
+            id="modal-verify-session-header"
+            className="font-bold text-xl text-center"
```
Comment (jaredhirsch) [inline packages/fxa-settings/src/components/ModalVerifySession/index.tsx:118]: How about adding a testid to this header?

File: packages/fxa-settings/src/components/ModalVerifySession/index.tsx [hunk 4]
```diff
@@ -109,26 +112,41 @@ export const ModalVerifySession = ({
...
+            email: <span className="font-bold"></span>,
+          }}
         >
-          Please enter the verification code that was sent to{' '}
-          <span className="font-bold">{primaryEmail.email}</span> within 5
-          minutes.
-        </p>
+          <p
+            id="modal-verify-session-desc"
+            data-testid="modal-desc"
```
Comment (jaredhirsch) [inline packages/fxa-settings/src/components/ModalVerifySession/index.tsx:133]: How about `modal-verify-session-desc`, so the testid will be unique to this modal

**Review by jaredhirsch — APPROVED** [review]
LGTM, two testid-related suggestions then good to land

**Pull request merged**